### DISCUSSION TITLE
Add support for remembering full screen and maximized state

### DISCUSSION THREAD

**Discussion by neilenns** [BetaHuhn/electron-win-state#188] [Ideas]
Currently this appears to only restore window position, and doesn't handle restoring whether the window was maximized or in full screen mode. It'd be great to have support for remembering those items as well.

**Comment by froschi123** [comment]
this is really something that would make it perfect!!!

**Comment by BetaHuhn** [comment]
Hi, thanks for the suggestion @neilenns!

I remember having issues with the maximize event listener in the past, I don't know if it didn't work at all or what the issue was though. That might be why it is not in this library now.

I will test it again and see what the issue was (if it existed at all) and hopefully get this added soon!

**Reply by neilenns** [reply]
I took care of the full screen state on my own, you may find the code useful. I made sure to set the window hidden to start so the user doesn't see a brief flash of the window at its default state before opening to full screen.

https://github.dev/neilenns/Collins-FMS-3000-Display/blob/bc1ec96273186d6e1d25d7dde5f6fec741286b62/src/main.js#L81-L98

And here's how I save the state:

https://github.dev/neilenns/Collins-FMS-3000-Display/blob/bc1ec96273186d6e1d25d7dde5f6fec741286b62/src/main.js#L107-L111

I didn't bother with the maximized state since for my little app that's not a mode people would normally run in.

**Reply by BetaHuhn** [reply]
Nice, thanks for sharing! TIL that you can link directly to the selected text in github.dev, that's pretty cool!

Feel free to create a PR yourself if you feel comfortable :)

**Reply by neilenns** [reply]
I might later today!
